src/async_pools/mongoose_async_pools.erl

```diff
@@ -0,0 +1,182 @@
+-module(mongoose_async_pools).
+
+-include("mongoose_config_spec.hrl").
+-include("mongoose_logger.hrl").
+
+-behaviour(supervisor).
+-export([start_link/3, init/1]).
+-ignore_xref([start_link/3]).
+
+% API
+-export([start_pool/3, stop_pool/2, config_spec/0]).
+-export([put_task/3, put_task/4]).
+-ignore_xref([put_task/3]).
+-export([sync/2]).
+
+-type task() :: term().
+-type pool_id() :: atom(). % The subsystem, like 'pm_mam', or 'inbox'
+-type pool_name() :: atom(). % The pool name, like 'inbox_sup_async_pool_localhost'
+-type pool_type() :: batch | aggregate.
+-type pool_opts() :: #{pool_type := pool_type(),
+                      _ => _}.
+-type worker_opts() :: #{host_type := mongooseim:host_type(),
+                         pool_id := pool_id(),
+                         flush_extra := pool_extra(),
+                         batch_size => pos_integer(),
+                         flush_interval => pos_integer(),
+                             flush_callback => flush_callback(),
+                              prep_callback => prep_callback(),
+                         aggregate_callcack => aggregate_callback(),
+                           request_callcack => request_callback()
```

Comment:
I see what you wanna do here with indention.
Could we just align it regularly? ;)